tide-log/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+#![feature(async_await)]
+#![deny(
+    nonstandard_style,
+    rust_2018_idioms,
+    future_incompatible,
+    missing_debug_implementations
+)]
```

Comment:
While this does seem to be the standard in this repo, I would suggest moving away from using `deny` in the crate as that just makes developers lives harder when they have temporary warnings being emitted during development. It's better to turn on extra lints via `warn` and then have CI include `--deny=warnings` to force developers to ensure code is warning free before it makes it to master (I know this is more of a policy question than specific to this PR, but the introduction of a new crate seems like a good time to bring up cross-crate standards like this).

Reply:
I agree! It followed along from the core. I think it makes perfect sense to do this only for the core, and let other crates have warnings instead. Is that what you had in mind?

PS. Could we leave `nonstandard_idioms` at deny though? Or would you prefer to leave everything as just warnings?

cc/ @yoshuawuyts @tirr-c

Reply:
Personally I do not believe _any_ crate should _ever_ use `deny` _in code_. I'm a very big proponent of using Clippy and enabling extra warnings that are `allow`-by-default, but those should be denied _in CI only_. From what I have seen the Rust community as a whole seems to be onboard with treating all warnings as issues that need fixing, so it's highly likely they will fix them before even opening a PR. If a developer doesn't then CI will fail and I have yet to see a developer that has complained about being forced to fix warnings in their PRs.

Denying warnings in code has some significant downsides:

 * When you are mid development it causes extra compile cycles because of trivial issues e.g. missing putting `#[derive(Debug)]` on a new struct. It's not uncommon to add code while developing that you later decide is not necessary and gets deleted, being forced to fix warnings in that code just so you can run your tests is a real pain.

 * It causes incompatibility with future Rust versions. The set of code that a particular lint applies to is not guaranteed to stay the same across Rust versions, if it expands to cover something in the current codebase then users that have a `path` dependency on Tide or are in some other way compiling it without `--cap-lints` will no longer be able to build their project.

Reply:
Thanks @Nemo157! This is an incredibly valuable insight! Personally, I tend to use Makefiles which fails on publish/release and handle all the complexity there so it's consistent between CI and developers. So, I'm already convinced on this. 

Considering the `forbid` pattern was in tide since early days, should we wait until everyone pitches in or possibly today's meeting?

Reply:
@prasannavl I'm empathetic to Nemo's point here, and :+1: on loosening the constraints here if it makes contributing easier (:

Reply:
Awesome! Let me do this as a separate PR and do a full sweep for all crates once #220 merges.

Reply:
Converted to issue: https://github.com/rustasync/tide/issues/232